File: C++/Leetcode/Array/jumpGame.cpp
#include <iostream>
#include <vector> 

class Solution{
    public:
        bool jumpGame(std::vector<int>& nums){
            int goalpost = nums.size()-1;

            for(int i=nums.size()-2; i >= 0; --i){
                if ((nums[i] + i) >= goalpost){
                    goalpost = i;
                }
            }
            return goalpost==0;
        }
};


int main(){
    Solution solution_instance;
    std::vector<int> nums = {2,3,1,1,4};
    std::cout << solution_instance.jumpGame(nums) << std::endl;
}

File: Python/Leetcode/Array/jumpGame.py
class Solution:
    def canJump(self, nums: list[int]) -> bool:
        goalpost = len(nums)-1
        
        for i in range(len(nums)-2, -1, -1):
            if nums[i] + i >= goalpost:
                goalpost = i
                
        return goalpost==0       
        
if __name__=='__main__':
    solution_instance = Solution()
    arr = [3,2,1,0,4]
    
    print(solution_instance.canJump(arr))
    